core/src/replay_stage.rs

```diff
@@ -3077,48 +3077,51 @@ impl ReplayStage {
                 }
 
                 if bank.collector_id() != my_pubkey {
-                    // If the block does not have at least DATA_SHREDS_PER_FEC_BLOCK shreds in the last FEC set,
-                    // mark it dead. No reason to perform this check on our leader block.
-                    if !blockstore
-                        .is_last_fec_set_full(bank.slot())
-                        .inspect_err(|e| {
+                    // If the block does not have at least DATA_SHREDS_PER_FEC_BLOCK correctly retransmitted
+                    // shreds in the last FEC set, mark it dead. No reason to perform this check on our leader block.
+                    let check_result = match blockstore.check_last_fec_set(bank.slot()) {
+                        Ok(last_fec_set_check_results) => {
+                            // Update metrics regardless of feature flag
+                            last_fec_set_check_results.report_metrics(bank_slot, bank.hash());
+                            // Get a final result based on the feature flags
+                            last_fec_set_check_results.get_result(&bank.feature_set)
```

Comment:
Any reason not to push this inside of `check_last_fec_set`? Seems like the caller shouldn't have to be aware of this

Reply:
good call, i've moved this into a blockstore wrapper. Also i've changed it to return the last fec set merkle root, aka the "block_id" for use in the future.